### PR TITLE
ADD instructions on how to set up a secure development environment.

### PR DESCRIPTION
**Pull request by Afrowave**
## Description

Facebook have moved their users to [secure by default](https://www.facebook.com/notes/facebook-engineering/secure-browsing-by-default/10151590414803920/#:~:text=We've%20focused%20on%20making,long%20used%20https%20as%20well.). 

## Rationale

If a developer chooses to build *Facebook* login capability with the `users` app, then setting `https` in the developer environment will be necessary.

If required, I can add this capability through a new PR so that it becomes part of the project configuration.

### CODE REVIEW

File: docs/developing-locally-docker.rst [hunk 1]
```diff
@@ -200,3 +199,96 @@ Prerequisites:
...
+
+You should allow the new hostname. ::
+
+  ALLOWED_HOSTS = ["localhost", "0.0.0.0", "127.0.0.1", "my-dev-env.local"]
+
+Rebuild your ``docker`` application. ::
+
+  $ docker-compose -f local.yml up -d --build
+
+Go to your browser and type in your usrl, ``
```
Comment (browniebroke) [inline docs/developing-locally-docker.rst:283]: Looks like something is missing here... Did you mean to write something like that?

```suggestion
Go to your browser and type in your URL bar ``https://my-dev-env.local``
```

File: docs/developing-locally-docker.rst [hunk 2]
```diff
@@ -200,3 +199,96 @@ Prerequisites:
...
+
+      # HTTPS
+      # ------------------------------------------------------------------------------
+      VIRTUAL_HOST=my-dev-env.local
+      VIRTUAL_PORT=8000
+
+   The services run behind the reverse proxy.
+
+config/settings/local.py
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (browniebroke) [inline docs/developing-locally-docker.rst:273]: There is a Sphinx warning on this line:

```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Afrowave** [comment]
Thanks @browniebroke for having a look at this.

**Comment by browniebroke** [comment]
As a side note for next time, could you please use [a feature branch (as opposed to using master in your fork)](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/) to send your pull request? That would avoid the extra commits that we can see in here, like the one from #2661.

**Comment by Andrew-Chen-Wang** [comment]
@browniebroke I don't believe this is necessary since django-allauth doesn't require social login capabilities to be in an HTTPS local machine environment ref #2759 which raised this methodology. I haven't tested a social login with Facebook's OAuth2 capabilities, but AFAIK with Stripe and other OAuth apps, this isn't necessary and the docs might be suggesting something that's not really suitable for the local development environment.

**Comment by Afrowave** [comment]
I am sorry, @browniebroke,

I meant to get round to this and fix it now that the docs run in a docker container. I will fix this and also write a blog post to this. I have given myself the task of amintaining this part of the docs.

I actually thought I had set up notifications. I hope that is okay with you.

**Pull request merged**